packages/framework/src/Model/Mail/Mailer.php

```diff
@@ -15,42 +16,30 @@ class Mailer
 {
     public const DISABLED_MAILER_DSN = 'null://null';
 
-    /**
-     * @var \Symfony\Component\Mailer\MailerInterface
-     */
-    protected MailerInterface $symfonyMailer;
-
-    /**
-     * @var \Shopsys\FrameworkBundle\Model\Mail\MailTemplateFacade
-     */
-    protected MailTemplateFacade $mailTemplateFacade;
-
-    /**
-     * @var \Psr\Log\LoggerInterface
-     */
-    protected LoggerInterface $logger;
-
     /**
      * @param \Symfony\Component\Mailer\MailerInterface $symfonyMailer
      * @param \Shopsys\FrameworkBundle\Model\Mail\MailTemplateFacade $mailTemplateFacade
      * @param \Psr\Log\LoggerInterface $logger
+     * @param \Shopsys\FrameworkBundle\Model\Mail\Setting\MailSettingFacade $mailSettingFacade
      */
     public function __construct(
-        MailerInterface $symfonyMailer,
-        MailTemplateFacade $mailTemplateFacade,
-        LoggerInterface $logger
+        protected readonly MailerInterface $symfonyMailer,
+        protected readonly MailTemplateFacade $mailTemplateFacade,
+        protected readonly LoggerInterface $logger,
+        protected readonly MailSettingFacade $mailSettingFacade,
     ) {
-        $this->symfonyMailer = $symfonyMailer;
-        $this->mailTemplateFacade = $mailTemplateFacade;
-        $this->logger = $logger;
     }
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Mail\MessageData $messageData
+     * @param int $domainId
      */
-    public function send(MessageData $messageData): void
+    public function send(MessageData $messageData, int $domainId): void
```

Comment:
Because of https://github.com/shopsys/shopsys/pull/2592#discussion_r1166260182 this method definition stay intact, but was deprecated in backward compatible way in favor `sendForDomain` which behave like suggested in mentioned comment. So old and new implementations works depending on which method is used.